spec/lib/rake/extensiontask_spec.rb

```diff
@@ -620,6 +620,102 @@
         spec.metadata['allowed_push_host'].should eq 'http://test'
       end
 
+      it "should set required_rubygems_version when building a gem for `-linux-gnu` or `-linux-musl`" do
+        if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
+          skip "rubygems 3.3.22+ only"
+        end
+
```

Comment:
Can we remove this and change expected value depending on RubyGems version instead?

```ruby
if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
  cross_specs["arm-linux-gnu"].required_rubygems_version.should eq(Gem::Requirement.default)
  cross_specs["arm-linux-musl"].required_rubygems_version.should eq(Gem::Requirement.default)
else
  cross_specs["arm-linux-gnu"].required_rubygems_version.should eq(expected_rubygems_version)
  cross_specs["arm-linux-musl"].required_rubygems_version.should eq(expected_rubygems_version)
end
```

Reply:
Please note that for earlier versions of Rubygems, `-linux-gnu` is not supported, and `-linux-musl` _might_ be supported, depending on the versions. So I've updated this spec to assert only on what we care about:

```ruby
        cross_specs["arm64-darwin"].required_rubygems_version.should eq(Gem::Requirement.default)
        cross_specs["arm-linux"].required_rubygems_version.should eq(Gem::Requirement.default)

        if Gem::Version.new(Gem::VERSION) >= Gem::Version.new("3.3.22")
          expected_rubygems_version = Gem::Requirement.new([">= 3.3.22"])
          cross_specs["arm-linux-gnu"].required_rubygems_version.should eq(expected_rubygems_version)
          cross_specs["arm-linux-musl"].required_rubygems_version.should eq(expected_rubygems_version)
        end
```

and it does not assert on what we don't care about (the possible values of the unsupported platforms' specs). I hope this is OK.

Reply:
I see.